10_Examples/jett-onboarding/src/testFile.cpp

```diff
@@ -0,0 +1,43 @@
+#include "gtest/gtest.h"
+#include "talking.h"
+#include "string.h"
+
+class TestExample : public ::testing::Test
+{
+protected:
+  TestExample()
+  {
+    talking_ = std::make_shared<talking>();
+  }
+
+  void SetUp() override
+  {
+  }
+
+  void TearDown() override
+  {
+  }
+
+  std::shared_ptr<talking> talking_;
+
+};
+
+TEST_F(TestExample, testTalk) {
+  //testing the talk function
+  EXPECT_EQ("Test", talking_->talk("Test"));
+}
+
+TEST_F(TestExample, testNumberPrint) {
+  EXPECT_TRUE(talking_->numberPrint() == talking_->num1);
```

Comment:
A few more facts on gtest.

You can have many different EXPECT_XXX statements in a single test (and many times you will). In the case where you want a test to completely give up if something fails, gtest has ASSERT_XXX(...). If youd like it to continue running the tests, you can use EXPECT_XXX. I usually default to EXPECT.